packages/amplify-provider-awscloudformation/src/graphql-resource-manager/amplify-graphql-resource-manager.ts

```diff
@@ -218,15 +234,13 @@ export class GraphQLResourceManager {
     return `${ROOT_APPSYNC_S3_KEY}/${buildHash}/states`;
   };
 
-  private gsiManagement = (diffs: DiffChanges<DiffableProject>, currentState: DiffableProject, nextState: DiffableProject) => {
-    const gsiChanges = _.filter(diffs, diff => {
-      return diff.path.includes('GlobalSecondaryIndexes');
-    });
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private gsiManagement = (diffs: DiffChanges<DiffableProject>, currentState: DiffableProject, nextState: DiffableProject): any => {
+    const gsiChanges = _.filter(diffs, diff => diff.path.includes('GlobalSecondaryIndexes'));
 
     const tableWithGSIChanges = _.uniqBy(gsiChanges, diff => diff.path?.slice(0, 3).join('/')).map(gsiChange => {
-      const tableName = gsiChange.path[3] as string;
-
-      const stackName = gsiChange.path[1].split('.')[0] as string;
+      const tableName = (gsiChange.path[0] === ROOT_LEVEL ? gsiChange.path[2] : gsiChange.path[3]) as string;
```

Comment:
can this be extracted in a function and reused at line 335 and 337?

Reply:
The logic is slightly different and also the return types. This doesn't benefit from moving to a different function.